nginx-sys/src/lib.rs

```diff
@@ -121,6 +136,15 @@ impl ngx_str_t {
         std::str::from_utf8(self.as_bytes()).unwrap()
     }
 
+    /// Create an `ngx_str_t` instance from a byte slice.
+    ///
+    /// # Safety
+    ///
+    /// The caller must provide a valid pointer to a memory pool.
+    pub unsafe fn from_bytes(pool: *mut ngx_pool_t, src: &[u8]) -> Option<Self> {
```

Comment:
`from_string()` and `from_str()`, both return Self, not `Option<Self>`, shouldn't we change them as well?

Reply:
Not sure if it's worth fixing the `str` conversion functions. I planned to remove these or at least mark as deprecated.